libs/shared/console-shared/src/lib/deploy-other-commit-modal/ui/deploy-other-commit-modal.tsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable-next-line */
+import { Commit } from 'qovery-typescript-axios'
+import {
+  Button,
+  ButtonSize,
+  ButtonStyle,
+  InputSearch,
+  LoaderSpinner,
+  TagCommit,
+  Truncate,
+  useModal,
+} from '@qovery/shared/ui'
+import { dateToFormat, timeAgo } from '@qovery/shared/utils'
+
+export interface DeployOtherCommitModalProps {
+  commitsByDay: Record<string, Commit[]>
+  isLoading: boolean
+  setSelectedCommitId: (commitId: string) => void
+  selectedCommitId: string | null
+  currentCommitId?: string
+  buttonDisabled?: boolean
+  handleDeploy: () => void
+  deployLoading: boolean
+  onSearch: (value: string) => void
+}
+
+export function DeployOtherCommitModal(props: DeployOtherCommitModalProps) {
+  const {
+    commitsByDay = {},
+    setSelectedCommitId,
+    selectedCommitId,
+    currentCommitId,
+    buttonDisabled = false,
+    handleDeploy,
+    deployLoading,
+    onSearch,
+  } = props
+  const { closeModal } = useModal()
+
+  return (
+    <div className="p-6">
+      <h2 className="h4 text-text-600 max-w-sm truncate mb-1">Deploy another version</h2>
+      <p className="mb-6 text-text-400 text-sm">Select the commit you want to deploy.</p>
+
+      <div className="mb-6">
+        <InputSearch placeholder="Search by commit message or commit id" onChange={onSearch} />
+      </div>
+
+      {props.isLoading && (
```

Comment:
We have a mix between `props.isLoading` and `const { } = props`, should maybe better to harmonize it ?